ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/ConversionsSpec.scala

```diff
@@ -93,6 +107,195 @@ class ConversionsSpec extends AnyWordSpec with Matchers {
       val key2 = deduplicationKeyBytesFor(List("alice", "bob"))
       key1 should not be key2
     }
+
+    "encode/decode rejections" should {
+      val submitterInfo = DamlSubmitterInfo.newBuilder().build()
+
+      def rejectionEntryIsConverted(rejection: Rejection, expectedCode: Code) = {
+        val encodedEntry = Conversions
+          .encodeTransactionRejectionEntry(
+            submitterInfo,
+            rejection,
+          )
+          .build()
+        Conversions
+          .decodeTransactionRejectionEntry(encodedEntry)
+          .value
+          .code shouldBe expectedCode.value()
+      }
+
+      "convert validation failure" in {
```

Comment:
Consider using `Table` for these tests, to make it terser and potentially clearer?

Reply:
Done, I'm usually not the biggest fan as it tends to make tests a lot more unreliable and I would take the extra boilerplate to have the test easier to read, but in this case I think it makes sense as the inputs are really simple